containers.go

```diff
@@ -395,6 +403,50 @@ func (r *containerStore) SetNames(id string, names []string) error {
 	return ErrContainerUnknown
 }
 
+func (r *containerStore) AddNames(id string, names []string) error {
+	if container, ok := r.lookup(id); ok {
+		for _, name := range container.Names {
+			delete(r.byname, name)
+			names = append(names, name)
+		}
+		names = dedupeNames(names)
+		for _, name := range names {
+			if otherContainer, ok := r.byname[name]; ok {
+				r.removeName(otherContainer, name)
+			}
+			r.byname[name] = container
+		}
+		container.Names = names
+		return r.Save()
```

Comment:
I am not entirely familiar with the code but I am surprised we're deleting something. I'd imagine something similar to:
```golang
container.Names = dedupeNames(append(containers.Names, names...))
```

Reply:
Scratch that: having read more code, we need to delete the previous references to other containers.

Reply:
@vrothberg Is there anything for me to do here ? or could you mark resolved.